### ISSUE TITLE
Cannot upload file to Storage. uploadTask.putFile callback is not called

### ISSUE THREAD

**Issue by kainjinez** [firebase/firebase-ios-sdk#1888]
### [REQUIRED] Step 2: Describe your environment

  * Xcode version: 10
  * Firebase SDK version: Newest from Git
  * Firebase Component: Storage (Auth, Core, Database, Firestore, Messaging, Storage, etc)
  * Component version: Newest from branch master

### [REQUIRED] Step 3: Describe the problem
I'm developing uploading screenshot to Storage for macOS application. I used these code below, it does not throw any error or crash. I don't see any new file on Storage after upload and callback at `putFile` won't be called.
#### Steps to reproduce:
```
        let storage = Storage.storage()
        let localFile = URL(string: "/Users/truonghieu/Library/Containers/com.example.appname/Data/Documents/appname/image.jpg")
        let storageRef = storage.reference().child("abc.jpg")
        let uploadTask = storageRef.putFile(from: localFile!, metadata: nil) { metadata, error in
            guard metadata != nil else{
                print("error: \(String(describing: error?.localizedDescription))")
                return
            }
            let fileManager = FileManager.default
            if fileManager.fileExists(atPath: localPath) {
                do {
                    try fileManager.removeItem(at: localFile!)
                    print("success: move item\(localPath)")
                }catch{
                    print("failed: move item\(localPath)")
                }
            }
        }

```

**Comment by paulb777** [comment]
@kainjinez We might be able to help you here, but usage questions are more appropriate for stackoverflow.

**Comment by kainjinez** [comment]
@paulb777  Hi Paul,
I use these code similar to what i see in Firebase Storage document. So I think this is maybe a bug or i need to do anything more to let it works, because this is a product of community, maybe~

**Comment by schmidt-sebastian** [comment]
Unfortunately, we don't currently have any debug logging in our Storage SDKs. Are you able to verify that we are able to connect to the backend (by monitoring the app's network traffic through Activity Monitor or similar?). How large is the file you are uploading?

**Comment by kainjinez** [comment]
@schmidt-sebastian Hi Sebastian,
For verifying connection, i am able to login/register using Firebase Auth.
My files is just a screenshot with 2MB max.
I tried to downgrade the Firebase SDK from 5.x to 4.8.0 and above code works.

**Comment by sohilmemon** [comment]
Please solve this issue! I am also facing the same. I have done all the changes and after 12 hours I came to know that the issue exists in SDK. Success / Error callback doesn't call!

**Comment by kainjinez** [comment]
@sohilmemon You may try to downgrade the SDK to this version, it works for me. Use this and wait for their fix 
https://drive.google.com/file/d/12jZl9xjP7KVOjIHGioHwctgZMYhvaFo5/view?usp=sharing

**Comment by schmidt-sebastian** [comment]
@sohilmemon Are you also using the MacOS version of the SDK?

**Comment by azeemse** [comment]
Did anyone find the solution to this issue? I am still facing this issue.

**Comment by kainjinez** [comment]
> Did anyone find the solution to this issue? I am still facing this issue using iOS SDK version **v5.7**

You can temporary pass through this issue by using older version. Check drive link above.

**Comment by schmidt-sebastian** [comment]
@kainjinez @azeemse @sohilmemon Does one of you have a minimal code sample that could reproduce this issue? I would like to drill down which release between 4.8 and 5.7 introduced this regression and having a hard time since the callbacks fire for me.

A good starting point would be the quickstart app: https://github.com/firebase/quickstart-ios/tree/master/storage

Do you see the same issues there?

What device are you encountering this issue on?

**Comment by h36ahmed** [comment]
Hi there, I am also finding this issue as well with the latest version of Firebase. I cannot upload file and there is no callback.

**Comment by kainjinez** [comment]
@schmidt-sebastian since i faced this problem when developing MacOS app, then I cannot check your quickstart because it belong to iPhone app. Even it works with iPhone, it may not related with this issue.

**Comment by h36ahmed** [comment]
Okay so update from me is that I believe I figured what was wrong with my code. I calling the api on another thread other than main and hence, I used `DispatchQueue.main.async { myCode }`and it worked like a charm. Hope this helps anyone. Nothing wrong with Firebase Storage code.

**Comment by azeemse** [comment]
> Since I have not heard any reports, I assume that downloads work fine?

I was using the latest version of Firebase for the first time in any of my project so I just followed the documentation.
Performed each and every step 1 by 1 and reached to the point where I needed to upload the profile image to Storage.

Copied the code written in documentation and pasted in my project to just check how it works and what it returns but the code never ran successfully. It executed and observes the "progress" status but never returns any "completion block" 

And most interestingly no help is available on Stack Overflow or any other platform. This thread gave me some clue that this is the issue of SDK so I tried @kainjinez solutions, turned down the version to 4.8.0 and it simply started working.


Things to remember, I am building an iOS application

**Xcode:**              10.0
**Technology:**     Swift 4.2
**Minimum Deployment Target:**   iOS v10.0

**Comment by azeemse** [comment]
> Okay so update from me is that I believe I figured what was wrong with my code. I was calling the api on another thread other than main and hence, I used `DispatchQueue.main.async { myCode }`and it worked like a charm. Hope this helps anyone. Nothing wrong with Firebase Storage code.

With the latest iOS SDK, I tried this solution but this didn't work for me. Details are mentioned in my above comment.

**Comment by schmidt-sebastian** [comment]
Thanks for all your help :) We already looked into moving the Storage internal operations off the main queue, and this thread is now another good reason to do so.  We will release a fix soon.

**Comment by azeemse** [comment]
> Thanks for all your help :) We already looked into moving the Storage internal operations off the main queue, and this thread is now another good reason to do so. We will release a fix soon.

Is this issue is resolved and Can I update my pods to the latest version now?

**Comment by schmidt-sebastian** [comment]
We are still working on the release. The fix will be out in the next couple of days.

**Comment by paulb777** [comment]
The [milestone](https://github.com/firebase/firebase-ios-sdk/milestone/19) associated with this issue will be closed when 5.12.0 is published.
